aspnetcore/fundamentals/middleware/index/snapshot/Chain/StartupUseWhen.cs

```diff
@@ -0,0 +1,31 @@
+﻿public class Startup
+{
+    private readonly ILogger<Startup> _logger;
+
+    public Startup(ILogger<Startup> logger)
+    {
+        _logger = logger;
+    }
+
+    private void HandleBranchAndRejoin(IApplicationBuilder app)
+    {
+        app.Use(async (context, next) =>
+        {
+            var branchVer = context.Request.Query["branch"];
+            _logger.LogInformation("Branch used = {branchVer}", branchVer);
+
+            await next();
```

Comment:
Should I add a comment here to be explicit about why we're logging and no longer trying to write to the response? It's already explained earlier in the topic, but might be useful here too. i.e.:

```csharp
// Do work that doesn't write to the Response.
await next();
```